google-cloud-pubsublite/src/test/java/com/google/cloud/pubsublite/internal/TopicStatsClientImplTest.java

```diff
@@ -0,0 +1,222 @@
+
+    } catch (StatusException e) {
+      throw e.getStatus().asRuntimeException();
+    }
+  }
+
+  private static Partition partition() {
+    try {
+      return Partition.of(0);
+    } catch (StatusException e) {
+      throw e.getStatus().asRuntimeException();
+    }
+  }
+
+  private static Offset start() {
+    return Offset.of(1);
+  }
+
+  private static Offset end() {
+    return Offset.of(2);
+  }
+
+  private static ComputeMessageStatsRequest request() {
+    return ComputeMessageStatsRequest.newBuilder()
+        .setTopic(path().toString())
+        .setPartition(partition().value())
+        .setStartCursor(Cursor.newBuilder().setOffset(start().value()).build())
+        .setEndCursor(Cursor.newBuilder().setOffset(end().value()).build())
+        .build();
+  }
+
+  private static ComputeMessageStatsResponse response() {
+    return ComputeMessageStatsResponse.newBuilder().setMessageBytes(1).setMessageCount(2).build();
+  }
+
+  private final TopicStatsServiceGrpc.TopicStatsServiceImplBase serviceImpl =
+      mock(
+          TopicStatsServiceGrpc.TopicStatsServiceImplBase.class,
+          delegatesTo(new TopicStatsServiceGrpc.TopicStatsServiceImplBase() {}));
+
+  private TopicStatsClientImpl client;
+
+  @Before
+  public void setUp() throws IOException {
+    String serverName = InProcessServerBuilder.generateName();
+    grpcCleanup.register(
+        InProcessServerBuilder.forName(serverName)
+            .directExecutor()
+            .addService(serviceImpl)
+            .build()
+            .start());
+    ManagedChannel channel =
+        grpcCleanup.register(InProcessChannelBuilder.forName(serverName).directExecutor().build());
+    TopicStatsServiceGrpc.TopicStatsServiceBlockingStub stub =
+        TopicStatsServiceGrpc.newBlockingStub(channel);
+    client =
+        new TopicStatsClientImpl(REGION, stub, TopicStatsClientSettings.DEFAULT_RETRY_SETTINGS);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    client.shutdownNow();
+    Preconditions.checkArgument(client.awaitTermination(10, SECONDS));
+  }
+
+  @Test
+  public void region_isConstructedRegion() {
+    assertThat(client.region()).isEqualTo(REGION);
+  }
+
+  private static <T> Answer<Void> answerWith(T response) {
+    return invocation -> {
+      StreamObserver<T> responseObserver = invocation.getArgument(1);
+      responseObserver.onNext(response);
+      responseObserver.onCompleted();
+      return null;
+    };
+  }
+
+  private static Answer<Void> answerWith(Status status) {
+    return invocation -> {
+      StreamObserver<?> responseObserver = invocation.getArgument(1);
+      responseObserver.onError(status.asRuntimeException());
+      return null;
+    };
+  }
+
+  private static Answer<Void> inOrder(Answer<Void>... answers) {
+    AtomicInteger count = new AtomicInteger(0);
+    return invocation -> {
+      int index = count.getAndIncrement();
+
+      return answers[index].answer(invocation);
+    };
+  }
```

Comment:
can you centralize these?

Reply:
I moved these to a static TestUtil class. Does that sound reasonable?